source/graphql/migrate-wundergraph.txt

```diff
@@ -0,0 +1,197 @@
+<https://cosmo-docs.wundergraph.com/>`__.
+
+Migrate to WunderGraph
+----------------------
+
+The WunderGraph BFF is an open source gateway that bundles up your data
+and generates type-safe clients for your frontend framework. If you
+plan on using GraphQL Federation, take a look at WunderGraph Cosmo.
+
+If you are starting from scratch, follow the `MongoDB / Atlas Data source
+documentation <https://docs.wundergraph.com/docs/wundergraph-config-ts-reference/configure-mongodb-atlas-data-source>`__.
+For more details, refer to the `WunderGraph docs page for Atlas 
+<https://docs.wundergraph.com/docs/supported-data-sources/mongodb-atlas>`__.
+
+You can also use hooks to add additional business logic, or `custom GraphQL 
+resolvers <https://docs.wundergraph.com/docs/features/custom-graphql-resolvers>`__
+to decouple the clients from the database.
+
+.. procedure::
+
+   .. step:: Create a WunderGraph Application
+
+      .. code-block:: shell
+         :copyable: true
+   
+         # Init a new project
+         npx create-wundergraph-app my-project --example simple
+
+         # Move to the project directory
+         cd my-project
+
+         # Install dependencies
+         npm i
+   
+   .. step:: Add npm Scripts
+
+      Add the following scripts to ``package.json``, so you can run the
+      WunderGraph server.
+
+      .. code-block:: shell
+         :copyable: true
+   
+         {
+            "scripts": {
+               "start": "wunderctl up --debug",
+               "build": "wunderctl generate"
+            }
+         }
+
+   .. step:: Start the WunderGraph Server
+
+      .. code-block:: shell
+         :copyable: true
+   
+         npm run start
+
+      After starting the server, WunderGraph does some code generation. Check
+      the server status at ``http://localhost:9991``.
+
+   .. step:: Configure MondoDB Atlas Data Source
+
+      You need to configure your MongoDB Atlas datasource. Inside your 
+      ``wundergraph.config.ts`` file add the following:
+
+      .. code-block:: shell
+         :copyable: true
+   
+         // wundergraph.config.ts
+         const Atlas = introspect.mongodb({
+            apiNamespace: 'my_db',
+            databaseURL: 'YourAtlasURL',
+         });
+
+         configureWunderGraphApplication({
+            apis: [Atlas],
+         });
+
+   .. step:: Add Query Operations
+
+      To work with your Atlas data, you need to add query operations. Navigate
+      to the ``operations`` folder and create a new GraphQL file called
+      ``Users.graphql`` and add the following: 
+
+      .. code-block:: shell
+         :copyable: true
+   
+         {
+            Mongo_findFirstusers {
+               id
+               name
+               email
+            }
+         }
+
+   .. step:: Browse Data with GraphiQL
+
+      After you get your WunderGraph server running and add query operations,
+      you can use the GraphiQL user interface to explore your Atlas data through
+      a GraphQL API.
```

Comment:
```suggestion
      a GraphQL API. This can be useful when your app is in development. For apps in production, we recommend using WunderGraph's JSON-RPC interface (see next step).
```